### PR TITLE
fix: trigger creation bug

### LINKED ISSUE #1911: Trigger names
**Is your feature request related to a problem? Please describe.**
Current naming of triggers is a bit confusing. 

image

**Describe the solution you'd like**

Screen Shot 2020-01-21 at 4 09 28 PM

### LINKED ISSUE #2158: How to create an **Intent recognized** trigger for a Regex-defined intent?
## Describe the bug
I was trying to update the weather tutorials and came up with this question: how to create an **Intent recognized** trigger and configure it to an intent defined using Regular Expressions? It seems users can only create an **Intent recognized** trigger using LUIS as recognizer type in the current new trigger wizard design implemented in [PR1786](https://github.com/microsoft/BotFramework-Composer/pull/1786). 
@luhan2017 and @hibrenda found the similar issue during our meeting yesterday.

## To Reproduce

1. Create an intent using Regular Expressions as a recognizer type. 
2. Create an **Intent recognized** trigger 
3. Have no idea to configure the intent defined using RegEx to this trigger

## Expected behavior

**Intent recognized** trigger should be created to 

## Screenshots

QQ截图20200303145930

### LINKED ISSUE #2193: Unable to create non OnIntent trigger
## Screenshots

https://share.getcloudapp.com/d5u01dRN

### PR DESCRIPTION
**Pull request by liweitian**
## Description
1. fix bug which block users to create some kinds of trigger. 
2. update regEx UI.
3. message received type is added

## Task Item
Closes #2193 
Closes #2158 
Closes #1911 

## Screenshots
aaaaa

### CODE REVIEW

**Comment by hibrenda** [comment]
#2158

**Comment by boydc2014** [comment]
can you add a gif? how did you add regex back?

**Comment by cwhitten** [comment]
nice work @liweitian & @alanlong9278 !

File: Composer/packages/client/src/utils/dialogUtil.ts [hunk 1]
```diff
@@ -73,23 +82,52 @@ export function insert(content, path: string, position: number | undefined, data
...
 
-  const insertAt = typeof position === 'undefined' ? current.length : position;
-
-  current.splice(insertAt, 0, newStep);
+export function generateRegexExpression(intent: string, pattern: string) {
+  return { intent, pattern };
+}
 
-  set(content, path, current);
+export function createNewTrigger(dialog: DialogInfo, data: TriggerFormData): DialogInfo {
```
Comment (zhixzhan) [inline Composer/packages/client/src/utils/dialogUtil.ts]: The return type `DialogInfo` I believe is an indexed dialog type, but here you didn't
 reindex it, and seems no need to reindex it. So I would suggest return `content: string`.

File: Composer/packages/client/src/utils/dialogUtil.ts [hunk 2]
```diff
@@ -73,23 +82,52 @@ export function insert(content, path: string, position: number | undefined, data
...
+  const dialogCopy = cloneDeep(dialog);
+  const trigger = generateNewTrigger(data);
+  insert(dialogCopy.content, 'triggers', undefined, trigger);
+  return dialogCopy;
+}
 
-  return content;
+export function createRegExIntent(dialog: DialogInfo, intent: string, pattern: string): DialogInfo {
```
Comment (zhixzhan) [inline Composer/packages/client/src/utils/dialogUtil.ts]: see above.

**Pull request merged**